x/gamm/keeper/swap.go

```diff
@@ -49,6 +49,7 @@ func (k Keeper) swapExactAmountIn(
 	}
 	tokensIn := sdk.Coins{tokenIn}
 
+	ctx.GasMeter().ConsumeGas(types.GasFeeForSwap, "swap computation")
```

Comment:
It feels safer to move the gas consumption to the top-most level of abstraction, i.e the `msg` handler for swapping.

Reply:
I guess really it should probably be defined per pool type

Reply:
That way we have it per Balancer swap, rather than per total swap. (And can then later charge a different gas amount for stableswap / next CFMM)

Reply:
Yeah that makes sense, just updated! Should we update balancer pool docs with this or should we wait until it becomes a param?